### PR TITLE
Werror fixes

### PR DESCRIPTION
**Pull request by pbanicev**
This PR contains mostly -Werror fixes for recent gcc and clang compilers.

### CODE REVIEW

File: engine/src/Conf/vb_engine_conf.c [hunk 1]
```diff
@@ -1202,7 +1202,7 @@ static t_VB_engineErrorCode VbEngineConfFileRead(const char *path)
   ezxml_t              socket_alive_conf;
   ezxml_t              user_profiles;
   ezxml_t              align_params;
-  struct in6_addr      ip_addr;
+  struct in6_addr      ip_addr = { 0 };
```
Comment (vgraumxl) [inline engine/src/Conf/vb_engine_conf.c:1205]: **This change triggers following compilation error with old GCC version (4.4):**

src/Conf/vb_engine_conf.c: In function \u2018VbEngineConfFileRead\u2019:
src/Conf/vb_engine_conf.c:1205:10: error: missing braces around initializer [-Werror=missing-braces]
src/Conf/vb_engine_conf.c:1205:10: error: (near initialization for \u2018ip_addr.__in6_u\u2019) [-Werror=missing-braces]
cc1: all warnings being treated as errors

**Review by vgraumxl — CHANGES_REQUESTED** [review]

File: engine/src/Conf/vb_engine_conf.c (hunk 1 above)
Comment (pbanicev) [reply engine/src/Conf/vb_engine_conf.c:1205]: Is there some reason for still using 4.4 gcc version? Also some compilers versions expects it only is single braces, and some others older version of compiler two braces which complicates solutions and makes it more complex, like checking exact compiler version.

Comment (vgraumxl) [reply engine/src/Conf/vb_engine_conf.c:1205]: We have some test infrastructure with old GCC version, so as far as I know, we need to maintain it.
On the other hand, can you try with "= { } "? , this is working for GCC 4.4. Can you try it with newer GCC version?

Comment (pbanicev) [reply engine/src/Conf/vb_engine_conf.c:1205]: yes, will try.

**Pull request merged**